design/client-secret-rotation.md

```diff
@@ -78,3 +83,13 @@ The changes needed in the Admin UI for a client would be:
 This approach allows a global control over client secrets within a realm, and has less changes to the APIs, and is fully
 backwards compatible as there would be no change to the default behaviour. It also fits better into dynamic client
 registration than the previous proposal.
+
+Admin notification events
+-------------------------
+
+Throw an administrative events related to rotation and expiration of secrets.
+- When a client rotates a secret. Client ID and Client Name / Description.
+- When trying to authenticate using an expired rotated secret.
+- When a _percentage_ or _fixed amount_ remains to expire the secret. At this point we can add a new setting to the secret rotation.
```

Comment:
I recall that we earlier discussed together with Stian to add support to the "contacts" as the configuration option of the client. This option is also supported in the OIDC specification https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata .

This will allow that when client authentication is done and there is too small amount of time to expire the client secret (Maybe specified by "Remaining rotation period" or some other way), we can send the email to the client administrator, who was specified in the clients "contacts" . Is it what you meant?

Reply:
But I think that this is probably something to consider as a follow-up (Milestone 2). We probably don't manage to add this to RHSSO 7.6 though...

Reply:
I confess I didn't remember this detail but it's exactly the point I want to get to. The initial idea was to be reactive by launching events from a certain point to notify administrators that possibly the secrets would expire within a time window. Regarding _"Remaining rotation period"_ this parameter was created to force rotation of dynamic clients. In this case for dynamic clients there will be no notification but the rotation directly. I think a new field is more correct because that way we can use it for common and dynamic clients.

Returning to the topic of **_contacts_**, I think it makes much more sense because we will be proactive. This should be a new activity that would serve as a prerequisite for the notifications theme, do you agree?

Certainly, we must consider the frequency of sending these emails, I believe that sending an email with each authentication that is about to expire makes the emails quickly considered SPAM.

Reply:
+1 for consider the frequency of sent emails somehow. I am not sure about re-using the existing option or use the existing one...

It will be probably good to start dedicated discussion about this - maybe it even deserves to do the presentation on keycloak-weekly call for this after RHSSO 7.6 release :-) We can discuss more details about this...